### PR TITLE
testing/freerdp2: new aport

### PR DESCRIPTION
**Pull request by vakartel**
A Remote Desktop Protocol Implementation (2nd branch)
http://www.freerdp.com

### CODE REVIEW

File: testing/freerdp2/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+pkgver=2.0.0_rc0
+_pkgver=${pkgver/_rc/-rc}
+pkgrel=0
+pkgdesc="A Remote Desktop Protocol Implementation"
+url="http://www.freerdp.com/"
+arch="all"
+options="!check"
+license="ASL-2.0"
+depends=""
+makedepends="cmake linux-headers alsa-lib-dev cups-dev eudev-dev gsm-dev
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: You should move the `-dev` stuff here to `depends_dev` where needed (e.g. if packages compiling against its headers also need the headers of those libraries), and include `$depends_dev` in `makedepends`.

File: testing/freerdp2/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+builddir="$srcdir/$_pkgname-$_pkgver"
+
+# secfixes:
+#   2.0.0-r0:
+#     - CVE-2017-2834
+#     - CVE-2017-2835
+
+build() {
+	cd "$builddir"
+	cmake -DCMAKE_BUILD_TYPE=Release \
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: Are all of these explicitly needed?

File: testing/freerdp2/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+		-DWITH_XCURSOR=ON \
+		-DWITH_XEXT=ON \
+		-DWITH_XKBFILE=ON \
+		-DWITH_XI=ON \
+		-DWITH_XINERAMA=ON \
+		-DWITH_XRENDER=ON \
+		-DWITH_XV=ON \
+		-DWITH_ZLIB=ON \
+		-DWITH_THIRD_PARTY=ON \
+		|| return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: `|| return 1` can be omitted with `set -e` being the default.

File: testing/freerdp2/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	make
+}
+
+package() {
+	cd "$builddir"
+	make DESTDIR="$pkgdir" install
+}
+
+libs() {
+	default_libs || return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: Likewise as above.

File: testing/freerdp2/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	rmdir "$pkgdir"/usr/lib
+}
+
+channels() {
+	pkgdesc="$pkgdesc (virtual channel extensions)"
+	mkdir -p "$subpkgdir"/usr/lib || return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: As above.

File: testing/freerdp2/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+	mv "$pkgdir"/usr/lib/freerdp2 \
+		"$subpkgdir"/usr/lib
+}
+
+_shadow() {
+	pkgdesc="$_pkgname utility for sharing a X display via RDP"
+	mkdir -p "$subpkgdir"/usr/bin || return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: As above.

File: testing/freerdp2/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+	mv "$pkgdir"/usr/bin/freerdp-shadow-cli \
+		"$subpkgdir"/usr/bin
+}
+
+_winpr() {
+	pkgdesc="$_pkgname Windows Portable Runtime tools"
+	mkdir -p "$subpkgdir"/usr/bin|| return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: As above.

File: testing/freerdp2/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+	mv "$pkgdir"/usr/bin/winpr-* \
+		"$subpkgdir"/usr/bin
+}
+
+_libs() {
+	local name=${subpkgname#$pkgname-libs-}
+	pkgdesc="$pkgdesc ($name libraries)"
+	cd "$pkgdir"
+	local file; for file in $(find ./usr/lib -name "*$name*"); do
+		mkdir -p "$subpkgdir"/${file%/*} || return 1
```
Comment (Shizmob) [inline testing/freerdp2/APKBUILD]: As above.

File: testing/freerdp2/APKBUILD (hunk 2 above)
Comment (vakartel) [inline testing/freerdp2/APKBUILD]: Ok, I removed -DCMAKE_BUILD_TYPE=Release because it's by default

File: testing/freerdp2/APKBUILD (hunk 1 above)
Comment (vakartel) [inline testing/freerdp2/APKBUILD]: as described in libfreerdp/freerdp2.pc, the only required 3rd-party library is libssl and it's included automatically to freerdp2-dev depends.

> depend = freerdp2-libs-client=2.0.0_rc0-r0
> depend = freerdp2-libs-server=2.0.0_rc0-r0
> depend = freerdp2-libs-shadow=2.0.0_rc0-r0
> depend = freerdp2-libs-winpr=2.0.0_rc0-r0
> depend = freerdp2-libs=2.0.0_rc0-r0
> depend = pc:libssl
> depend = pkgconfig

So I see no sense to add all those -dev stuff to depends_dev.

File: testing/freerdp2/APKBUILD (hunk 3 above)
Comment (vakartel) [inline testing/freerdp2/APKBUILD]: done and with all below too

**Pull request closed without merging**